File: Algorithms/1337. The K Weakest Rows in a Matrix.rs
impl Solution {
    pub fn k_weakest_rows(mat: Vec<Vec<i32>>, k: i32) -> Vec<i32> {
        let mat = mat.iter().map(|r| r.iter().sum()).collect::<Vec<i32>>();
        let mut result = (0..(mat.len() as i32)).collect::<Vec<i32>>();
        result.sort_by_key(|&k| mat[k as usize]);

        result[..(k as usize)].to_vec()
    }
}

File: Algorithms/1337. The K Weakest Rows in a Matrix.ts
function kWeakestRows(mat: number[][], k: number): number[] {
  const sum = (nums: number[]) => nums.reduce((a, c) => a + c, 0);
  const calculatedMat = mat.map((row, i) => [i, sum(row)]);
  calculatedMat.sort((a, b) => a[1] - b[1]);
  return calculatedMat.map((e) => e[0]).slice(0, k);
}
